include/CL/opencl.hpp

```diff
@@ -3395,6 +3395,9 @@ class Event : public detail::Wrapper<cl_event>
     static cl_int
     waitForEvents(const vector<Event>& events)
     {
+        static_assert(sizeof(cl::Event) == sizeof(cl_event),
+        "Size of cl::Event must be equal to size of cl_event");
```

Comment:
Formatting nitpick (it would be great to find some clang-format settings that work for this file eventually):

```suggestion
        static_assert(sizeof(cl::Event) == sizeof(cl_event),
            "Size of cl::Event must be equal to size of cl_event");
```

Reply:
Hi Ben, thanks for comment.
This PR was come as a result of your last comment (https://github.com/KhronosGroup/OpenCL-CLHPP/pull/199#discussion_r1131198010).

I am agree that putting assert everywhere where events.front() is using, is overkill, so I decided to put this in one place.
Maybe you are right that static_assert in one of the methods of Event class is a little odd, but from other hand it also uses this kind of casting and it is closly place where unexpecting changes could happen. 

If you still will want to move this assert, I can put in this part of code: https://github.com/KhronosGroup/OpenCL-CLHPP/blob/main/include/CL/opencl.hpp#L3453
What do you think ?

Reply:
I'm fine with this as-is.  We can always move it later...